File: typescript/sylbreak.ts
const myConsonant: string = "\u1000-\u1021"; // "က-အ"

const enChar: string = "a-zA-Z0-9";

// "ဣဤဥဦဧဩဪဿ၌၍၏၀-၉၊။!-/:-@[-`{-~\s"
const otherChar: string = "\u1023\u1024\u1025\u1026\u1027\u1029\u102a\u103f\u104c\u104d\u104f\u1040-\u1049\u104a\u104b!-/:-@\\[-`\\{-~\\s";

const ssSymbol: string = "\u1039";

const ngaThat: string = "\u1004\u103a";

const aThat: string = "\u103a";

// Regular expression pattern for Myanmar syllable breaking
// *** a consonant not after a subscript symbol AND a consonant is not
// followed by a-That character or a subscript symbol
export const MYANMAR_SYLLABLE_BREAK_PATTERN: RegExp = new RegExp("((?!" + ssSymbol + ")[" + myConsonant + "](?![" + aThat + ssSymbol + "])" + "|[" + enChar + otherChar + "])", "mg");

export class Sylbreak {
    
    static segment(unicode: string): string[] {
        if (!unicode) {
            return new Array();
        }

        var outArray = unicode.replace(MYANMAR_SYLLABLE_BREAK_PATTERN, "𝕊$1").split('𝕊');
        if (outArray.length > 0) {
            outArray.shift();
        }
        return outArray;
    }

    static segmentWithSeparator(unicode: string, separator: string): string {
        if (!unicode) {
            return "";
        }
        if (!separator) {
            separator = "|";
        }
        return unicode.replace(MYANMAR_SYLLABLE_BREAK_PATTERN, separator + "$1");
    }
}

File: perl/sylbreak.pl
#!/usr/bin/perl

## syllable breaking tool for Myanmar language
## usage: ./sylbreak.pl <-i filename>  [-s separator] [-p {0} or 1]
## e.g. usage1: ./sylbreak.pl -i ../data/my-input
##      usage2: cat ../data/my-input2 | ./sylbreak.pl
##      usage3: ./sylbreak.pl -i ../data/my-input -s "/" -p=1
##
## last updated: 22 July 2016
## added space cleaning parts: 23 May 2022
## Author: Ye Kyaw Thu, Visiting Researcher, Waseda University
## HP:https://sites.google.com/site/yekyawthunlp/

## Reference of Myanmar Unicode: http://unicode.org/charts/PDF/U1000.pdf

use strict;
use warnings;
use utf8;
use Getopt::Long qw(GetOptions);

binmode STDIN, ":encoding(UTF-8)";
binmode STDOUT, ":encoding(UTF-8)";

my $iOption; #input filename
my $sOption; my $sep = "\|"; #separator
my $pOption; my $printInput = 0; #default value false for print option
my $fh;

GetOptions(
   'help|h|?' => \&help,
   'input-file|i=s' => \$iOption,
   'separator|s=s' => \$sOption,
   'print|p=i' => \$pOption,
) or die "Usage: ./syl-RE-break.pl <-i filename>  [-s separator] [-p {0} or 1]\n";


if ($iOption)
{
  open($fh, '<:encoding(UTF-8)', $iOption) or die "Could not open file '$iOption $!";
}elsif (!defined $iOption )
{
  $fh = *STDIN;
}
else
{
   help();
}


my $myConsonant = "က-အ";
my $enChar = "a-zA-Z0-9";
my $otherChar = "ဣဤဥဦဧဩဪဿ၌၍၏၀-၉၊။!-\/:-\@\[-`{-~\\s";
my $ssSymbol = "္";
my $aThat = "်";


$sep = $sOption if defined ($sOption);
$printInput = $pOption if defined ($pOption);

sub help 
{
    print "Usage: ./syl-RE-break.pl <-i filename>  [-s separator] [-p {0} or 1]\n";
    exit(0);
}


while (my $line = <$fh>)
{
   chomp $line;
   if ($printInput == 0)
   {
      
      #Regular expression pattern for Myanmar syllable breaking
      #*** a consonant not after a subscript symbol AND a consonant is not followed by a-That character or a subscript symbol
      $line =~ s/((?<!$ssSymbol)[$myConsonant](?![$aThat$ssSymbol])|[$enChar$otherChar])/$sep$1/g;
      $line =~ s/^\s+|\s+$//g;
      $line =~ s/ +/ /g;      
      print "$line\n";
   }elsif ($printInput == 1)
   {
      print "input: $line\n";
      $line =~ s/((?<!$ssSymbol)[$myConsonant](?![$aThat$ssSymbol])|[$enChar$otherChar])/$sep$1/g;
      $line =~ s/^\s+|\s+$//g;
      $line =~ s/ +/ /g;      
      print "output: $line\n";
   }

}

close($fh); 
